File: backend/app/schemas/student.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import date, datetime
from uuid import UUID

class StudentBase(BaseModel):
    student_id: str
    roll_number: Optional[str] = None
    full_name: str
    class_id: Optional[UUID] = None
    section: Optional[str] = None
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    parent_guardian: Optional[str] = None
    contact: Optional[str] = None
    status: str = 'active'
    face_enrolled: bool = False

class StudentCreate(StudentBase):
    pass

class StudentUpdate(BaseModel):
    student_id: Optional[str] = None
    roll_number: Optional[str] = None
    full_name: Optional[str] = None
    class_id: Optional[UUID] = None
    section: Optional[str] = None
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    parent_guardian: Optional[str] = None
    contact: Optional[str] = None
    status: Optional[str] = None
    face_enrolled: Optional[bool] = None

class StudentInDB(StudentBase):
    id: UUID
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/student.ts
// Student types aligned with backend schemas (app/schemas/student.py)

export interface Student {
  id: string;
  student_id: string;
  roll_number: string | null;
  full_name: string;
  class_id: string | null;
  section: string | null;
  date_of_birth: string | null;
  gender: string | null;
  parent_guardian: string | null;
  contact: string | null;
  status: string;
  face_enrolled: boolean;
  created_at: string;
  updated_at: string;
}

export interface StudentCreate {
  student_id: string;
  roll_number?: string;
  full_name: string;
  class_id?: string;
  section?: string;
  date_of_birth?: string;
  gender?: string;
  parent_guardian?: string;
  contact?: string;
  status?: string;
  face_enrolled?: boolean;
}

export interface StudentUpdate {
  student_id?: string;
  roll_number?: string;
  full_name?: string;
  class_id?: string;
  section?: string;
  date_of_birth?: string;
  gender?: string;
  parent_guardian?: string;
  contact?: string;
  status?: string;
  face_enrolled?: boolean;
}
